pkg/apb/provision_or_update.go

```diff
@@ -49,11 +52,17 @@ func provisionOrUpdate(
 		return "", nil, errors.New("No image field found on instance.Spec")
 	}
 
+	k8scli, err := clients.Kubernetes(log)
+	if err != nil {
+		log.Error("Something went wrong getting kubernetes client")
+		return "", nil, err
+	}
+
 	ns := instance.Context.Namespace
-	log.Info("Checking if project %s exists...", ns)
-	if !projectExists(ns) {
-		log.Errorf("Project %s does NOT exist! Cannot provision requested %s", ns, instance.Spec.FQName)
-		return "", nil, fmt.Errorf("Project %s does not exist", ns)
+	log.Info("Checking if namespace %s exists.", ns)
+	_, err = k8scli.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return "", nil, fmt.Errorf("Project %s does NOT exist!", ns)
```

Comment:
One more :wink: 
```
pkg/apb/provision_or_update.go:65:30: error strings should not be capitalized or end with punctuation or a newline
```

You can use `make check` to run through most of the tests before pushing.

Reply:
I'll move back to what was there :).